nix/sources.nix

```diff
@@ -0,0 +1,194 @@
+# This file has been generated by Niv.
```

Comment:
if this is generated, should it be committed?

Reply:
Good question. I asked our fellow nixers and there are two good arguments for it:
- What's the workflow of not committing? Running `niv` to regenerate it on the go?
- The above would be non-reproducible, since `sources.nix` may slightly change across `niv` upgrades

So commit I have. ;)

Reply:
ok, makes sense.

Still, I think it makes sense to mention how this was generated (which command was run, at which point of the setup), so that if people wanted to recreate this for their own project, they have a step-by-step guide

Reply:
Hm, I think I'm misunderstanding the project a little bit then. Is this supposed to be a template or a reference implementation?

Reply:
a bit of both, I guess! Some users would just copy and paste, some other would try to recreate it changing things here and there. I don't see how hiding how this file was generated could help understanding how to setup a similar project

Reply:
Added more info inside the file.